python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -448,9 +446,14 @@ def predict(self, X, deep_check=False):
         ----------
         X : dask cuDF dataframe or numpy array (n_samples, n_features)
         deep_check : boolean (default = False)
```

Comment:
@canonizer comment in the C++ side made me think a similar thing. Why is this needed? Is there a chance the result is wrong? Is this just for debugging or testing? Now it is clear what it does, but not why its needed or its intended purpose/use

Reply:
This is a check so that if we update our code or treelite updates their code or if dues to some other reason the information handoff gets messed up this will make sure that the forest created after concatenation is right. As that is the most important step for using FIL with mnmg RF, I added a check here. It should also help make debugging easier for us in the future wrt mnmg rf predict.

Since I had added this check in, I made it available to the user so they can check the concatenated forest created in case they face some errors.

Reply:
I think if we want to catch problems in the information hand-off, it's best to have tests for that, either in C++ or in Python.

Reply:
Absolutely agreed with @canonizer, and also it just seems super weird and implies that we can return wrong error to the users and that is an ok behavior! Guaranteed correct results is what we should aim for in genaral, at least as much as stochastic, convergence and precision allows things. 

This type of variable for debugging like this should **not** be part of the API as is. The fact that it makes things easier for us to debug is not a good reason to add it to the user facing API at all, it is an implementation thing that a user should not need to worry about. At best using **kwargs to hide it or have a `debug` parameter instead (still far from ideal, but better than it is). But still, this really looks bad in my opinion, almost looking as if we are telling the user "if things are wrong on our results, you can go ahead and test it for us" which sounds terrible, or at least it looks that way the way the code is right now.